tests/test-faceting.php

```diff
@@ -5,58 +5,61 @@
-		$this->factory->post->create( array( 'post_title' => 'tags-b-and-c', 'tags_input' => array( 'tag-b', 'tag-c' ), 'post_date' => '2010-09-01 00:00:00' ) );
-		$this->factory->post->create( array( 'post_title' => 'tags-a-and-c', 'tags_input' => array( 'tag-a', 'tag-c' ), 'post_date' => '2010-10-01 00:00:00' ) );
-
-		$this->parent_one = $this->factory->post->create( array( 'post_title' => 'parent-one', 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:01' ) );
-		$this->parent_two = $this->factory->post->create( array( 'post_title' => 'parent-two', 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:02' ) );
-		$this->parent_three = $this->factory->post->create( array( 'post_title' => 'parent-three', 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:03' ) );
-		$this->factory->post->create( array( 'post_title' => 'child-one', 'post_parent' => $this->parent_one, 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:04' ) );
-		$this->factory->post->create( array( 'post_title' => 'child-two', 'post_parent' => $this->parent_one, 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:05' ) );
-		$this->factory->post->create( array( 'post_title' => 'child-three', 'post_parent' => $this->parent_two, 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:06' ) );
-		$this->factory->post->create( array( 'post_title' => 'child-four', 'post_parent' => $this->parent_two, 'post_type' => 'page', 'post_date' => '2007-01-01 00:00:07' ) );
-
-		// Force refresh the index so the data is available immediately
-		SP_API()->post( '_refresh' );
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+
+		$posts_to_index = array();
+
+		$author_a = self::factory()->user->create( array( 'user_login' => 'author_a', 'user_pass' => rand_str(), 'role' => 'author' ) );
+		$posts_to_index[] = self::factory()->post->create( array( 'post_title' => rand_str(), 'post_author' => $author_a, 'post_date' => '2007-01-04 00:00:00' ) );
+		$posts_to_index[] = self::factory()->post->create( array( 'post_title' => rand_str(), 'post_author' => $author_a, 'post_date' => '2007-01-05 00:00:00' ) );
+
+		$author_b = self::factory()->user->create( array( 'user_login' => 'author_b', 'user_pass' => rand_str(), 'role' => 'author' ) );
+		$posts_to_index[] = self::factory()->post->create( array( 'post_title' => rand_str(), 'post_author' => $author_b, 'post_date' => '2007-01-03 00:00:00' ) );
+
+		$cat_a = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'cat-a' ) );
+		$cat_b = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'cat-b' ) );
+		$cat_c = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'cat-c' ) );
+
+		$posts_to_index[] = self::factory()->post->create( array( 'post_title' => 'cats-a-b-c', 'post_date' => '2008-12-01 00:00:00', 'post_category' => array( $cat_a, $cat_b, $cat_c ) ) );
```

Comment:
This is very similar to what is being done in `class-searchpress-unit-test-case.php`. Possible to reuse any of that, or abstract it? There's alternately an argument for keeping it visible and verbosity being a strength.

Reply:
it's definitely possible, but the data is ever-so-slightly different so it requires rewriting some of the tests below. absolutely worth doing, just a task for another day. Created #124 to ensure that doesn't get lost.